assembly/decoder.ts

```diff
@@ -301,22 +313,41 @@ export class JSONDecoder<JSONHandlerT extends JSONHandler> {
   }
 
   private parseNumber(): bool {
-    // TODO: Parse floats
-    let number: i64 = 0;
-    let sign: i64 = 1;
+    let number: f64 = 0;
+    let sign: f64 = 1;
+    let decimalPlace: f64 = 1;
     if (this.peekChar() == "-".charCodeAt(0)) {
       sign = -1;
       this.readChar();
     }
     let digits = 0;
-    while (CHAR_0 <= this.peekChar() && this.peekChar() <= CHAR_9) {
-      let byte = this.readChar();
-      number *= 10;
-      number += byte - CHAR_0;
-      digits++;
+    while (CHAR_PERIOD == this.peekChar() || (CHAR_0 <= this.peekChar() && this.peekChar() <= CHAR_9)) {
+      if (CHAR_PERIOD == this.peekChar()) {
+        // Lower the decimal place
+        decimalPlace = decimalPlace / 10;
+
+        // Read the character to continue reading
+        this.readChar();
+      } else {
+        let byte = this.readChar();
+        let value: f64 = byte - CHAR_0;
+        if (decimalPlace == 1) {
+          number *= 10;
+          number += value;
+        } else {
+          value *= decimalPlace;
+          number += value;
+          decimalPlace = decimalPlace / 10;
+        }
+        digits++;
+      }
```

Comment:
I recommend read whole float and use `parseFloat` method due to this custom parser will be very imprecise